marketing/sailthruScriptJob.groovy

```
@@ -0,0 +1,130 @@
+
+
+/*
+Example secret YAML file used by this script
+
+    oauth_host : The host used to obtain Course Discovery access token
+    oauth_key : Key used to obtain Course Discovery access token
+    oauth_secret : Secret used to obtain Course Discovery access token
+    sailthru_key : Access key for Sailthru api
+    sailthru_secret : Access secret for Sailthru api
+    content_api_url : Url of Course Discovery API
+    lms_url : Url of LMS (default http://courses.edx.org
+    report_email : Email address to sent batch report to
+    type : A choice of "programs", "courses" or "all" 
+    fixups : any fix
+    mode : command mode
+    requirements : path to sailthru requirements file in repo
+    scriptPath : the script to run for this job
+    githubRepoUrlSegment : url segment of repo i.e edx/repo_name
+    jobName : job-name-to-be-created
+    notificationEmail : Post Action to send email for job notification 
+*/
+
+/* stdout logger */
+/* use this instead of println, because you can pass it into closures or other scripts. */
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
+
+/* Map to hold the k:v pairs parsed from the secret file */
+Map secretMap = [:]
+out.println('Parsing secret YAML file')
+try {
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    String contents = new File("${EDX_SAILTHRU_SYNC_CONTENT}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(contents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    assert jobConfig.containsKey('oauth_host')
```

Why bother with this? If something's missing, why not let the job blow up?